iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -682,6 +724,100 @@ static LogicalResult setRootConfig(
+/// transposition to use CPUDoubleTilingExpert pipeline.
+static LogicalResult setTransposeLikeOpRootConfig(
+    func::FuncOp entryPointFn, linalg::GenericOp genericOp,
+    ArrayRef<LoopTilingAndDistributionInfo> tiledLoops) {
+  if (getLoweringConfig(genericOp)) {
+    return success();
+  }
+
+  // If there are no loops, there is nothing to do.
+  unsigned numLoops = genericOp.getNumLoops();
+  if (numLoops == 0) {
+    return success();
+  }
+
+  if (!hasAVX2Features(genericOp) || !isSupportedTransposeOp(genericOp)) {
+    return success();
+  }
+
+  SmallVector<int64_t> minTileSizes =
+      getMinTilingSizesForEachDim(entryPointFn, genericOp);
+  SmallVector<int64_t> maxTileSizes(numLoops, defaultWorkgroupTileSize);
+  if (llvm::all_of(minTileSizes, [](int64_t vs) { return vs == 1; })) {
+    // Nothing to vectorize just lower to loops.
+    return success();
+  }
+
+  if (llvm::count_if(minTileSizes,
+                     [](int64_t tileSize) { return tileSize > 1; }) != 2) {
+    // Transpose patterns are not applicable if vectorizing more or less than
+    // two dims.
+    return success();
+  }
+
+  // Make sure that the original tile sizes are multiple of the tile sizes
+  // to be used for the transpose op (i.e., 8x8).
+  // TODO(diegocaballero): Enable 4x8 tile sizes if we find it useful.
+  if (llvm::any_of(minTileSizes, [](int64_t tileSize) {
+        return tileSize > 1 && (tileSize % 8) != 0;
+      })) {
+    return success();
+  }
+
+  // Replace dims to be vectorized with the new 8x8 tile sizes.
+  std::replace_if(
+      minTileSizes.begin(), minTileSizes.end(),
+      [](int64_t tileSize) { return tileSize > 1; }, 8);
+
+  // Set the flow level tiling to the default.
+  OpBuilder builder(genericOp.getContext());
+  builder.setInsertionPoint(genericOp);
+  auto linalgOp = cast<linalg::LinalgOp>(genericOp.getOperation());
+  SmallVector<Range> iterationDomain =
+      linalgOp.createLoopRanges(builder, genericOp.getLoc());
+  auto partitionableLoopsInterfaceOp =
+      cast<IREE::Flow::PartitionableLoopsInterface>(genericOp.getOperation());
+  SmallVector<int64_t> flowTileSizes = getDefaultDistributedLevelTileSizes(
+      iterationDomain, partitionableLoopsInterfaceOp, minTileSizes,
+      maxTileSizes);
+
+  // Set the next level tile sizes.
+  SmallVector<int64_t> parallelTileSizes;
+  SmallVector<int64_t> reductionTileSizes(numLoops, 0);
+  setX86WorkgroupTileSizes(genericOp, numLoops, flowTileSizes, minTileSizes,
+                           maxTileSizes, parallelTileSizes);
+
+  TileSizesListType tileSizes;
+  tileSizes.push_back(flowTileSizes);
+  tileSizes.push_back(parallelTileSizes);
+  tileSizes.push_back(reductionTileSizes);
+
+  // For non-tensor based ops use the Buffer ops pipeline.
+  auto passPipeline =
+      genericOp.hasTensorSemantics()
+          ? DispatchLoweringPassPipeline::CPUDoubleTilingExpert
+          : DispatchLoweringPassPipeline::CPUBufferOpsTileAndVectorize;
```

Comment:
I think transpose op won't got with earlier bufferization path, so we don't have to check tensor semantics.

@MaheshRavishankar please confirm this.

Reply:
Thats right. Transpose ops will not get bufferized during the early bufferization of copy only dispatches (if that was the question, sorry if I misread it)

Reply:
Actually, I mispoke here. Just found out that in case of rank-reducing subviews, a transposing generic op could be created by the pass that bufferizes copy only dispatches. So maybe it is OK for this logic to still be around.

Reply:
Ok, I'll undo that change and commit it for now. We can revisit this as needed.
Thanks for the feedback!